core/src/storage2/collections/bitstash/counts.rs

```diff
@@ -70,17 +70,30 @@ impl CountFree {
     ///
     /// Returns `None` if all counts are `0xFF`.
     pub fn position_first_zero(&mut self) -> Option<u8> {
-        for (i, count) in self.counts.iter_mut().enumerate() {
-            if !self.full.is_full(i as u8) {
-                if *count == !0 {
-                    self.full.set_full(i as u8);
-                } else {
-                    *count += 1;
-                }
-                return Some(i as u8)
-            }
+        let i = (!self.full.0).leading_zeros();
+        if i == 32 {
+            return None
+        }
+
+        Some(i as u8)
+    }
+
+    /// Increases the number of set bits for the given index.
+    ///
+    /// # Panics
+    ///
+    /// - If the given index is out of bounds.
+    /// - If the increment would cause an overflow.
+    pub fn inc(&mut self, index: usize) {
+        assert!(index < 32, "index is out of bounds");
+        if self.counts[index] == !0 {
+            self.full.set_full(index as u8);
+        } else {
+            let new_value = self.counts[index]
+                .checked_add(1)
+                .expect("set bits increment overflowed");
+            self.counts[index] = new_value;
```

Comment:
This is doubly checking the addition since we already have a check for `== !0` in the if. Not needed.